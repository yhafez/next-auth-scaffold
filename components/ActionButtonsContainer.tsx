// Path: ./components/ActionButtonsContainer.tsx
import { ReactNode } from 'react'
import { Box } from '@mui/material'

export interface ActionButtonsContainerProps {
	name: string
	children: ReactNode
}

export default function ActionButtonsContainer({ name, children }: ActionButtonsContainerProps) {
	return (
		<Box
			id={`${name}-buttons-container`}
			sx={{
				display: 'flex',
				flexDirection: 'row',
				justifyContent: 'space-around',
				alignItems: 'center',
				gap: 2,
				margin: '8px 4px',
				width: '80%',
			}}
		>
			{children}
		</Box>
	)
}
