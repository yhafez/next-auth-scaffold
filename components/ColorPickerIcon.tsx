// Path: ./components/ColorPicker.tsx
import { useState, MouseEvent, useCallback, useEffect } from 'react'
import { Box, IconButton, Typography } from '@mui/material'
import { ColorLens } from '@mui/icons-material'
import { darken, lighten } from 'color2k'

import { ColorPickerPopover } from '.'
import { useBoundStore } from '../store'
import {
	getContrastColor,
	getSecondaryColor,
	mixColors,
	findOptimalOverlayOpacity,
	isOverlayNecessary,
} from '../utils/helpers'

export interface ColorPickerProps {
	name: string
	selectedInit?: boolean
}

export default function ColorPickerIcon({ name, selectedInit }: ColorPickerProps) {
	const { darkMode, customPalette, setCustomPalette } = useBoundStore()

	const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null)
	const [selected, setSelected] = useState(selectedInit ?? false)

	const handleColorChange = useCallback((color: string) => {
		const contrast = getContrastColor(color)
		let mixedColorPrimaryMain = color
		if (isOverlayNecessary(color, contrast, 7)) {
			const overlayColor = contrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(contrast, overlayColor, color, 7)
			mixedColorPrimaryMain = mixColors(color, overlayColor, overlayOpacity)
		}
		let mixedColorPrimaryLight = lighten(mixedColorPrimaryMain, 0.2)
		if (isOverlayNecessary(mixedColorPrimaryLight, contrast, 7)) {
			const overlayColor = contrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(
				contrast,
				overlayColor,
				mixedColorPrimaryLight,
				7,
			)
			mixedColorPrimaryLight = mixColors(mixedColorPrimaryLight, overlayColor, overlayOpacity)
		}
		let mixedColorPrimaryDark = darken(mixedColorPrimaryMain, 0.2)
		if (isOverlayNecessary(mixedColorPrimaryDark, contrast, 7)) {
			const overlayColor = contrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(
				contrast,
				overlayColor,
				mixedColorPrimaryDark,
				7,
			)
			mixedColorPrimaryDark = mixColors(mixedColorPrimaryDark, overlayColor, overlayOpacity)
		}

		const secondaryColor = getSecondaryColor(mixedColorPrimaryMain)
		const secondaryColorContrast = getContrastColor(secondaryColor)
		let mixedColorSecondaryMain = secondaryColor
		if (isOverlayNecessary(secondaryColor, secondaryColorContrast, 7)) {
			const overlayColor = secondaryColorContrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(
				secondaryColorContrast,
				overlayColor,
				secondaryColor,
				7,
			)
			mixedColorSecondaryMain = mixColors(secondaryColor, overlayColor, overlayOpacity)
		}
		let mixedColorSecondaryLight = lighten(mixedColorSecondaryMain, 0.2)
		if (isOverlayNecessary(mixedColorSecondaryLight, secondaryColorContrast, 7)) {
			const overlayColor = secondaryColorContrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(
				secondaryColorContrast,
				overlayColor,
				mixedColorSecondaryLight,
				7,
			)
			mixedColorSecondaryLight = mixColors(mixedColorSecondaryLight, overlayColor, overlayOpacity)
		}
		let mixedColorSecondaryDark = darken(mixedColorSecondaryMain, 0.2)
		if (isOverlayNecessary(mixedColorSecondaryDark, secondaryColorContrast, 7)) {
			const overlayColor = secondaryColorContrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(
				secondaryColorContrast,
				overlayColor,
				mixedColorSecondaryDark,
				7,
			)
			mixedColorSecondaryDark = mixColors(mixedColorSecondaryDark, overlayColor, overlayOpacity)
		}

		const newPalette = {
			...customPalette,
			primary: {
				main: mixedColorPrimaryMain,
				contrastText: contrast,
				light: mixedColorPrimaryLight,
				dark: mixedColorPrimaryDark,
			},
			secondary: {
				main: mixedColorSecondaryMain,
				contrastText: secondaryColorContrast,
				light: mixedColorSecondaryLight,
				dark: mixedColorSecondaryDark,
			},
		}

		setCustomPalette(newPalette)
	}, [])

	useEffect(() => {
		const customPalette = localStorage.getItem('customPalette')
		if (customPalette) handleColorChange(customPalette)
	}, [handleColorChange])

	useEffect(() => {
		setSelected(selectedInit ?? false)
	}, [selectedInit])

	const open = Boolean(anchorEl)
	const id = open ? 'simple-popover' : undefined

	const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
		setAnchorEl(event.currentTarget)
	}

	const handleClose = () => {
		setSelected(false)
		setAnchorEl(null)
	}

	return (
		<>
			<Box
				id={`${name}-color-picker-container`}
				sx={{
					display: 'flex',
					flexDirection: 'column',
					alignItems: 'center',
					justifyContent: 'center',
					p: 0,
					height: '100%',
					width: '100%',
					cursor: 'pointer',
				}}
			>
				<IconButton
					id={`${name}-color-picker-button`}
					aria-describedby={id}
					aria-labelledby={`${name}-color-picker-label`}
					onClick={handleClick}
				>
					<ColorLens
						id={`${name}-color-picker-icon`}
						sx={{
							color: selected
								? darkMode
									? 'primary.light'
									: 'primary.dark'
								: 'primary.contrastText',
							fontSize: '1.5rem',
						}}
						aria-labelledby={`${name}-color-picker-label`}
						role="img"
					/>
				</IconButton>
				<label htmlFor={`${name}-color-picker-button`}>
					<Typography
						id={`${name}-color-picker-label`}
						variant="body2"
						sx={{
							textAlign: 'center',
							fontWeight: 500,
							cursor: 'pointer',
							color: selected
								? darkMode
									? 'primary.light'
									: 'primary.dark'
								: 'primary.contrastText',
						}}
					>
						Theme Color
					</Typography>
				</label>
			</Box>
			<ColorPickerPopover
				name={name}
				anchorEl={anchorEl}
				handleClose={handleClose}
				anchorOrigin={{
					vertical: 'bottom',
					horizontal: 'center',
				}}
				transformOrigin={{
					vertical: 'top',
					horizontal: 'center',
				}}
				handleColorChange={handleColorChange}
			/>
		</>
	)
}
