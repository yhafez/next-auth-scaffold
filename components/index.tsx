// Path: ./components/index.tsx
import Layout from './Layout'

import DashboardDrawer from './Drawers/DashboardDrawer'
import SettingsDrawer from './Drawers/SettingsDrawer'

import ColorPickerPopover from './Popovers/ColorPickerPopover'
import DrawerSignOutPopover from './Popovers/DrawerSignOutPopover'

import ActionButtonsContainer from './ActionButtonsContainer'
import ColorPickerIcon from './ColorPickerIcon'
import ConfirmPasswordInput from './ConfirmPasswordInput'
import DarkModeSwitch from './DarkModeSwitch'
import Drawer from './Drawer'
import DrawerButton from './DrawerButton'
import EditableTextField from './EditableTextField'
import EmailInput from './EmailInput'
import Modal from './Modal'
import ModalButton from './ModalButton'
import ModalNote from './ModalNote'
import NavigationButton from './NavigationButton'
import PasswordInput from './PasswordInput'
import SubmitButton from './SubmitButton'
import VisibilityIconButton from './VisibilityIconButton'

export {
	Layout,
	DashboardDrawer,
	SettingsDrawer,
	ColorPickerPopover,
	DrawerSignOutPopover,
	ActionButtonsContainer,
	ColorPickerIcon,
	ConfirmPasswordInput,
	DarkModeSwitch,
	Drawer,
	DrawerButton,
	EditableTextField,
	EmailInput,
	Modal,
	ModalButton,
	ModalNote,
	NavigationButton,
	PasswordInput,
	SubmitButton,
	VisibilityIconButton,
}
