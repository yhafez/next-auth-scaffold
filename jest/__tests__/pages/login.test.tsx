// Path: ./jest/__tests__/pages/login.test.tsx
import { render, act, screen, waitFor } from '@testing-library/react'
import { axe } from 'jest-axe'
import { SessionProvider } from 'next-auth/react'
import { ThemeProvider } from '@mui/material/styles'

import Login from '../../../pages/login'
import { Layout } from '../../../components'
import { defaultTheme } from '../../../theme'

describe('Login', () => {
	let container: HTMLElement

	beforeEach(() => {
		act(() => {
			container = render(
				<SessionProvider
					session={{
						token: {
							user: {
								id: '1',
								email: 'test@example.com',
								emailVerified: null,
								image: 'test',
								isDeactivated: false,
								name: null,
								role: 'USER',
								createdAt: new Date(),
								updatedAt: new Date(),
								isNewUser: false,
							},
						},
						expires: '1d',
					}}
				>
					<ThemeProvider theme={defaultTheme}>
						<Layout name="test" pageTitle="test">
							<Login />
						</Layout>
					</ThemeProvider>
				</SessionProvider>,
			).container

			return container
		})
	})

	it('Should have no accessibility violations', async () => {
		waitFor(async () => expect(await axe(screen.getByRole('main'))).toHaveNoViolations())
	})

	it('Should match snapshot', () => {
		expect(container).toMatchSnapshot()
	})
})
