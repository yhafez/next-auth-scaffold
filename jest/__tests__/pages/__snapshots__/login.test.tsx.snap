// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Should match snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-14umlvy-MuiButtonBase-root-MuiButton-root"
    id="test-skip-nav-link"
    tabindex="0"
    type="button"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-3so7zb-MuiTypography-root"
      id="test-skip-nav-link-text"
    >
      Skip to main content
    </p>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
  <div
    class="MuiBox-root css-114gqq5"
    id="test-layout-container"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-38ln4h-MuiPaper-root-MuiAppBar-root"
      id="test-app-bar"
    >
      <nav
        aria-label="header navigation"
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1l1du5z-MuiToolbar-root"
        id="test-toolbar"
      >
        <a
          aria-label="Home"
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1fqdykp-MuiTypography-root-MuiLink-root"
          href="/"
          id="test-toolbar-title-link"
          title="Home"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1 MuiTypography-noWrap css-63r4vv-MuiTypography-root"
            id="test-toolbar-title"
          >
            Next Auth Scaffold
          </h1>
        </a>
        <ul
          class="MuiList-root MuiList-padding css-166iuh2-MuiList-root"
          id="test-toolbar-icons"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-16ztiij-MuiListItem-root"
            id="test-toolbar-icons-color-picker-item"
          >
            <div
              class="MuiBox-root css-1lo2k45"
              id="test-toolbar-color-picker-container"
            >
              <button
                aria-labelledby="test-toolbar-color-picker-label"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                id="test-toolbar-color-picker-button"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  aria-labelledby="test-toolbar-color-picker-label"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1kqeszy-MuiSvgIcon-root"
                  data-testid="ColorLensIcon"
                  focusable="false"
                  id="test-toolbar-color-picker-icon"
                  role="img"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <label
                for="test-toolbar-color-picker-button"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-7xs8rl-MuiTypography-root"
                  id="test-toolbar-color-picker-label"
                >
                  Theme Color
                </p>
              </label>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-9gmft3-MuiListItem-root"
            id="test-toolbar-icons-dark-mode-switch-item"
          >
            <div
              class="MuiBox-root css-f9wnac"
              id="test-toolbar-dark-mode-switch-container"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium css-41alfi-MuiSwitch-root"
              >
                <span
                  class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-28iv8m-MuiButtonBase-root-MuiSwitch-switchBase"
                >
                  <input
                    aria-checked="false"
                    aria-labelledby="test-toolbar-dark-mode-switch-label-light"
                    class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                    id="test-toolbar-dark-mode-switch"
                    role="switch"
                    type="checkbox"
                  />
                  <svg
                    aria-hidden="false"
                    aria-labelledby="test-toolbar-dark-mode-switch-label-light"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-aq4ggj-MuiSvgIcon-root"
                    data-testid="WbSunnyIcon"
                    focusable="false"
                    id="test-toolbar-dark-mode-switch-icon-unchecked"
                    role="img"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m6.76 4.84-1.8-1.79-1.41 1.41 1.79 1.79 1.42-1.41zM4 10.5H1v2h3v-2zm9-9.95h-2V3.5h2V.55zm7.45 3.91-1.41-1.41-1.79 1.79 1.41 1.41 1.79-1.79zm-3.21 13.7 1.79 1.8 1.41-1.41-1.8-1.79-1.4 1.4zM20 10.5v2h3v-2h-3zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6zm-1 16.95h2V19.5h-2v2.95zm-7.45-3.91 1.41 1.41 1.79-1.8-1.41-1.41-1.79 1.8z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                />
              </span>
              <label
                for="test-toolbar-dark-mode-switch"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-191g3cx-MuiTypography-root"
                  id="test-toolbar-dark-mode-switch-label-light"
                >
                  Light
                  <br />
                   Mode
                </p>
              </label>
            </div>
          </li>
        </ul>
      </nav>
    </header>
    <main
      class="MuiBox-root css-cn70eu"
      id="test-layout-content"
      tabindex="-1"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-14umlvy-MuiButtonBase-root-MuiButton-root"
        id="login-skip-nav-link"
        tabindex="0"
        type="button"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-3so7zb-MuiTypography-root"
          id="login-skip-nav-link-text"
        >
          Skip to main content
        </p>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiBox-root css-114gqq5"
        id="login-layout-container"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-38ln4h-MuiPaper-root-MuiAppBar-root"
          id="login-app-bar"
        >
          <nav
            aria-label="header navigation"
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1l1du5z-MuiToolbar-root"
            id="login-toolbar"
          >
            <a
              aria-label="Home"
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1fqdykp-MuiTypography-root-MuiLink-root"
              href="/"
              id="login-toolbar-title-link"
              title="Home"
            >
              <h1
                class="MuiTypography-root MuiTypography-h1 MuiTypography-noWrap css-63r4vv-MuiTypography-root"
                id="login-toolbar-title"
              >
                Next Auth Scaffold
              </h1>
            </a>
            <ul
              class="MuiList-root MuiList-padding css-166iuh2-MuiList-root"
              id="login-toolbar-icons"
            >
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-16ztiij-MuiListItem-root"
                id="login-toolbar-icons-color-picker-item"
              >
                <div
                  class="MuiBox-root css-1lo2k45"
                  id="login-toolbar-color-picker-container"
                >
                  <button
                    aria-labelledby="login-toolbar-color-picker-label"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                    id="login-toolbar-color-picker-button"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      aria-labelledby="login-toolbar-color-picker-label"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1kqeszy-MuiSvgIcon-root"
                      data-testid="ColorLensIcon"
                      focusable="false"
                      id="login-toolbar-color-picker-icon"
                      role="img"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <label
                    for="login-toolbar-color-picker-button"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-7xs8rl-MuiTypography-root"
                      id="login-toolbar-color-picker-label"
                    >
                      Theme Color
                    </p>
                  </label>
                </div>
              </li>
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-9gmft3-MuiListItem-root"
                id="login-toolbar-icons-dark-mode-switch-item"
              >
                <div
                  class="MuiBox-root css-f9wnac"
                  id="login-toolbar-dark-mode-switch-container"
                >
                  <span
                    class="MuiSwitch-root MuiSwitch-sizeMedium css-41alfi-MuiSwitch-root"
                  >
                    <span
                      class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-28iv8m-MuiButtonBase-root-MuiSwitch-switchBase"
                    >
                      <input
                        aria-checked="false"
                        aria-labelledby="login-toolbar-dark-mode-switch-label-light"
                        class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                        id="login-toolbar-dark-mode-switch"
                        role="switch"
                        type="checkbox"
                      />
                      <svg
                        aria-hidden="false"
                        aria-labelledby="login-toolbar-dark-mode-switch-label-light"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-aq4ggj-MuiSvgIcon-root"
                        data-testid="WbSunnyIcon"
                        focusable="false"
                        id="login-toolbar-dark-mode-switch-icon-unchecked"
                        role="img"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="m6.76 4.84-1.8-1.79-1.41 1.41 1.79 1.79 1.42-1.41zM4 10.5H1v2h3v-2zm9-9.95h-2V3.5h2V.55zm7.45 3.91-1.41-1.41-1.79 1.79 1.41 1.41 1.79-1.79zm-3.21 13.7 1.79 1.8 1.41-1.41-1.8-1.79-1.4 1.4zM20 10.5v2h3v-2h-3zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6zm-1 16.95h2V19.5h-2v2.95zm-7.45-3.91 1.41 1.41 1.79-1.8-1.41-1.41-1.79 1.8z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                    />
                  </span>
                  <label
                    for="login-toolbar-dark-mode-switch"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-191g3cx-MuiTypography-root"
                      id="login-toolbar-dark-mode-switch-label-light"
                    >
                      Light
                      <br />
                       Mode
                    </p>
                  </label>
                </div>
              </li>
            </ul>
          </nav>
        </header>
        <main
          class="MuiBox-root css-cn70eu"
          id="login-layout-content"
          tabindex="-1"
        >
          <div
            class="MuiBox-root css-1bghj2m"
            id="login-modal-container"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-118nrrt-MuiTypography-root"
              id="login-modal-title-word-0"
            >
              Login
            </p>
            <div
              class="MuiBox-root css-7n00rz"
              id="login-modal-notification-container"
            />
            <form
              class="MuiBox-root css-xo10vp"
              id="login-modal-form"
              role="form"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-8v3hey-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-7yavjh-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="login-email-input"
                  id="login-email-input-label"
                >
                  Email
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-8r7mze-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="login-email-input"
                    type="email"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Email
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiTextField-root css-8v3hey-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-7yavjh-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="login-password-input"
                  id="login-password-input-label"
                >
                  Password
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-7kjzy7-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    id="login-password-input"
                    type="password"
                    value=""
                  />
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-dpfpu4-MuiButtonBase-root-MuiIconButton-root"
                    id="login-password-input-show-hide-button"
                    tabindex="0"
                    type="button"
                  >
                    <div
                      class="MuiBox-root css-erqh4x"
                      id="login-password-box"
                    >
                      <svg
                        aria-hidden="true"
                        aria-label="show password"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17slstj-MuiSvgIcon-root"
                        data-testid="VisibilityOffIcon"
                        focusable="false"
                        id="login-password-input-visibility-off-icon"
                        role="button"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                        />
                      </svg>
                      <span
                        class="MuiTypography-root MuiTypography-caption css-5oi5yi-MuiTypography-root"
                        id="login-password-input-show-password-text"
                      >
                        Show
                      </span>
                    </div>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Password
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiBox-root css-ueupuo"
                id="login-buttons-container"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-13dwpwp-MuiButtonBase-root-MuiButton-root"
                  id="login-signup-navigation-button"
                  tabindex="0"
                  type="button"
                >
                  Sign up
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1p925i6-MuiButtonBase-root-MuiButton-root"
                  id="login-submit-button"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiTypography-root MuiTypography-button css-11gc47n-MuiTypography-root"
                    id="login-submit-button-text"
                  >
                    Log in
                  </span>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div
                class="MuiBox-root css-3uapvq"
                id="login-forgot-password-container"
              >
                <label
                  for="remember-me-checkbox"
                  id="login-forgot-password-label"
                  style="display: flex; flex-direction: row; justify-content: center; align-items: center; gap: 1px; color: rgb(255, 255, 255); cursor: pointer; font-weight: 600;"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-18ye0vo-MuiTypography-root"
                    id="login-forgot-password-text"
                  >
                    Remember Me
                  </p>
                  <span
                    class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary css-1piw5z4-MuiButtonBase-root-MuiCheckbox-root"
                  >
                    <input
                      aria-checked="true"
                      checked=""
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate="false"
                      id="remember-me-checkbox"
                      role="checkbox"
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                </label>
              </div>
              <div
                class="MuiBox-root css-1h5af1d"
                id="login-forgot-password-modal-note-container"
              >
                <a
                  aria-label="Forgot password?"
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ptl0h6-MuiTypography-root-MuiLink-root"
                  href="/forgot-password"
                  id="login-forgot-password-modal-note-link"
                  title="Forgot password?"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-1qwu16v-MuiTypography-root"
                    id="login-forgot-password-modal-note-link-text"
                  >
                    Forgot password?
                  </p>
                </a>
              </div>
            </form>
          </div>
        </main>
      </div>
    </main>
  </div>
</div>
`;
